core/mavlink_parameters.cpp

```diff
@@ -111,7 +111,7 @@ MAVLinkParameters::do_work()
         _state = State::SET_PARAM_BUSY;
 
         char param_id[PARAM_ID_LEN] = {};
-        STRNCPY(param_id, set_param_work->param_name.c_str(), sizeof(param_id));
+        STRNCPY(param_id, set_param_work->param_name.c_str(), sizeof(param_id)-1);
```

Comment:
I don't get that :-/

Reply:
Oh, forget it, that's in the commit description :sweat_smile:

Reply:
GCC 8 doesn't like the fact that we could potentially copy 17 chars without null termination into `param_id`. At that point `param_id` would not be null terminated. That's how I understand it.

Reply:
I was just thinking, why don't we a hire a local `std::string` for doing this. Applicable elsewhere in such scenarios as well. Makes sense ?

Reply:
@shakthi-prashanth-m I don't think I understand what you mean.